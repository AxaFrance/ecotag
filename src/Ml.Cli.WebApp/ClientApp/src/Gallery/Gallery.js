import React from "react";

const Gallery = ({parentState}) => {
    
    console.log(parentState);
    
    return(
        <div>
            {parentState.files.map(file => {
                return(
                    <img src={URL.createObjectURL(file)} alt={file}/>
                )
            })}
        </div>
    )
};

export default Gallery;