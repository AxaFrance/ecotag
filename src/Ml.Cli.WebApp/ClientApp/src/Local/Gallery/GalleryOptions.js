import React, {useState} from "react";
import {Textarea} from "@axa-fr/react-toolkit-form-input-textarea";
import {SelectBase} from "@axa-fr/react-toolkit-form-input-select";
import Button from '@axa-fr/react-toolkit-button';
import './GalleryOptions.scss';

const optionsSelect = [
    {value: 'Recent to old', label: 'Recent to old'},
    {value: 'Old to recent', label: 'Old to recent'},
    {value: 'Alphabetic desc', label: 'Alphabetic desc'},
    {value: 'Alphabetic asc', label: 'Alphabetic asc'}
];

const sizeOptionsSelect = [
    {value: "64px", label: '64px'},
    {value: "128px", label: '128px'},
    {value: "256px", label: '256px'},
    {value: "512px", label: '512px'}
];

const GalleryOptions = ({state, setState, onSubmit}) => {
    
    const [filterState, setFilterState] = useState({
        filesPath: "",
        sortName: "Recent to old",
        size: "128px"
    });
    
    const applyOptions = () => {
        setState({
            ...state,
            sortName: filterState.sortName,
            size: filterState.size
        });
        onSubmit(filterState.filesPath);
    }
    
    return(
        <div className="gallery__header">
            <div className="gallery__options-container-left">
                <p className="tabs__title">Directory name:</p>
                <Textarea
                    id="text_area_stringsMatcherModifier"
                    name="TextAreaStringsMatcherModifier"
                    value={filterState.filesPath}
                    onChange={e => {
                        setFilterState({
                            ...filterState,
                            filesPath: e.value
                        });
                    }}
                />
                <p className="tabs__title">Sort by:</p>
                <SelectBase
                    id="select_type"
                    name="SelectType"
                    value={filterState.sortName}
                    options={optionsSelect}
                    onChange={e => {
                        setFilterState({...filterState, sortName: e.value});
                    }}
                />
                <p className="tabs__title">Size:</p>
                <SelectBase
                    id="select_type"
                    name="SelectType"
                    value={filterState.size}
                    options={sizeOptionsSelect}
                    onChange={e => {
                        setFilterState({...filterState, size: e.value});
                    }}
                />
            </div>
            <div className="gallery__options-container-right">
                <Button onClick={() => applyOptions()}>Apply filters</Button>
            </div>
        </div>
    )
};

export default GalleryOptions;