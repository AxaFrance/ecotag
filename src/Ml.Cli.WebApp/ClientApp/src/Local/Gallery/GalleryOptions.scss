.gallery{
  &__header{
    display: flex;
    background-color: #00008f;
    margin-bottom: 0.5em;
    color: white;
    justify-content: space-between;
  }
  &__options-container-left{
    display: flex;
    margin: 1em;
    height: 50px;
  }
  &__options-container-right{
    display: flex;
    margin: 1em;
    height: 50px;
  }
}