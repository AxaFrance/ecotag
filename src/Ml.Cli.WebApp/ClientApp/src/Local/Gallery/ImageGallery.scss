.image-gallery {

  &__container {
    margin: 1em;
    display: flex;
    flex-wrap: wrap;
  }

  &__link {
    border: 1px solid black;
    border-radius: 1em;
    padding: 1em;
    margin: 1em;
  }

  &__filename-container {
    width: 100px;
  }

  &__filename {
    width: 100%;
    overflow-wrap: break-word;
    text-align: center;
  }

  &__image-container {
    text-align: center;
  }

  &__image {
    position: relative;
    max-width: 100px;
    max-height: 100px;
  }
}

.filename-width {
  &__small {
    width: 64px;
  }

  &__medium {
    width: 128px;
  }

  &__normal {
    width: 256px;
  }

  &__big {
    width: 512px;
  }
}

.image-width {
  &__small {
    width: 64px;
    max-height: 64px;
  }

  &__medium {
    width: 128px;
    max-height: 128px;
  }

  &__normal {
    width: 256px;
    max-height: 256px;
  }

  &__big {
    width: 512px;
    max-height: 512px;
  }
}