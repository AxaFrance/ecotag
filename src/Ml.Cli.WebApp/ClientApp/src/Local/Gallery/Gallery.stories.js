import React from "react";
import Gallery from "./Gallery";
import {HashRouter} from "react-router-dom";

export default {
    title: 'Gallery',
    component: Gallery
};

const fetch = () => {};

const Template = (args) => <HashRouter basename={window.location.pathname}><Gallery {...args}/></HashRouter>;

export const Default = Template.bind({});
Default.args = {
    fetchFunction: fetch
};