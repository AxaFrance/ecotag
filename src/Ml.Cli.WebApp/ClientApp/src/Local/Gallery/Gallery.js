import React, {useEffect, useState, useRef} from "react";
import {fetchGetData, utf8_to_b64} from "../../FetchHelper";
import {ToastContainer} from "react-toastify";
import {Header, Name} from "@axa-fr/react-toolkit-layout-header";
import logo from '@axa-fr/react-toolkit-core/dist/assets/logo-axa.svg';
import {QueryClient, QueryClientProvider} from "react-query";
import ImageGallery from "./ImageGallery";
import GalleryOptions from "./GalleryOptions";
import TitleBar from "../TitleBar/TitleBar";
import {resilienceStatus} from './withResilience';

const queryClient = new QueryClient();

export const getFilesInfo = async data => {
    if(data.status === 200) {
        const hardDriveLocations = await data.json();
        let returnedList = [];
        hardDriveLocations.forEach(file => returnedList.push({name: file, url: `api/files/${utf8_to_b64(file)}`}));
        return returnedList;
    } else {
        return {};
    }
}

const loadStateAsync = (fetch) => async (setState, state, filesPath) => {
    setState({
        ...state,
        status: resilienceStatus.LOADING,
    });
    const uri = encodeURI("/api/gallery/" + filesPath);
    const response = await fetchGetData(fetch)(uri);
    if (response.status >= 300) {
        setState({
            ...state,
            status: resilienceStatus.ERROR,
            firstStatus: resilienceStatus.EMPTY,
        });
        return;
    }
    const filesList = await getFilesInfo(response);
    setState({
        ...state,
        files: filesList,
        status: resilienceStatus.EMPTY,
        firstStatus: resilienceStatus.EMPTY,
    });
};

function useInterval(callback, delay) {
    const savedCallback = useRef();

    // Remember the latest function.
    useEffect(() => {
        savedCallback.current = callback;
    }, [callback]);

    // Set up the interval.
    useEffect(() => {
        function tick() {
            savedCallback.current();
        }
        if (delay !== null) {
            let id = setInterval(tick, delay);
            return () => clearInterval(id);
        }
    }, [delay]);
}

const Gallery = ({fetchFunction}) => {
    
    //https://axafrance.visualstudio.com/Kubernetes/_git/dailyclean-api?path=%2Fweb%2Fsrc%2FApiStateProvider.js&version=GBmain
    //En bas: solution au problème de sauvegarde du state
    
    const [state, setState] = useState({
        files: [],
        filesPath: "",
        sortName: "Recent to old",
        size: "128px",
        status: resilienceStatus.EMPTY,
        firstStatus: resilienceStatus.EMPTY,
    });

    useInterval(() => {
        if(state.status !== resilienceStatus.LOADING && state.filesPath !== ""){
            loadStateAsync(fetchFunction)(setState, state, state.filesPath);
        }
    }, state.status === resilienceStatus.ERROR ? 15000: 5000);
    
    return(
        <QueryClientProvider client={queryClient}>
            <Header>
                <Name
                    title="ML-CLI"
                    subtitle="Made by AXA"
                    img={logo}
                    alt="AXA Logo"
                />
            </Header>
            <TitleBar title="Image Gallery" classModifier="af-title-bar--no-space"/>
            <GalleryOptions
                state={state}
                setState={setState}
            />
            <ImageGallery
                parentState={state}
            />
            <ToastContainer/>
        </QueryClientProvider>
    )
    
};

export default Gallery;