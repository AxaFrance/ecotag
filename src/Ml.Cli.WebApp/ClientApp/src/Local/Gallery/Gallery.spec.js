import Gallery, {sortItems} from "./Gallery";
import {act, fireEvent, render, waitFor} from '@testing-library/react';
import React from "react";
import {BrowserRouter as Router} from "react-router-dom";

jest.setTimeout(10000);

const returnedFiles =  [
    {date: "2021-07-02T17:07:15.0575353+02:00", file: "C:\\someFolder\\compare-licenses-file-1.json"},
    {date: "2021-08-02T17:07:15.0575353+02:00", file: "C:\\someFolder\\compare-licenses-file-2.json"},
    {date: "2021-09-02T17:07:15.0575353+02:00", file: "C:\\someFolder\\someImage.png"},
    {date: "2021-10-02T17:07:15.0575353+02:00", file: "C:\\someFolder\\somePdf.pdf"}
];

const sortFiles = [
    {date: "2021-07-02T17:07:15.0575353+02:00", name: "C:\\someFolder\\compare-licenses-file-1.json"},
    {date: "2021-08-02T17:07:15.0575353+02:00", name: "C:\\someFolder\\compare-licenses-file-2.json"},
    {date: "2021-09-02T17:07:15.0575353+02:00", name: "C:\\someFolder\\someImage.png"},
    {date: "2021-10-02T17:07:15.0575353+02:00", name: "C:\\someFolder\\somePdf.pdf"}
]

let nbFetchCalls = 0;

const fetch = () => {
    nbFetchCalls++;
    return {
        ok: true,
        status: 200,
        json: () => Promise.resolve(returnedFiles)
    };
};

const fetchKO = () => {
    return {ok: false, status: 400, json: () => Promise.resolve([])};
}

describe('Check images display', () => {
    test('Should render Gallery page correctly, resize images and sort by names', async () => {
        const {container, asFragment} = render(<Router basename="/"><Gallery fetchFunction={fetch} /></Router>);
        
        expect(asFragment()).toMatchSnapshot();
        
        const fileTextArea = container.querySelector(".af-form__input-text");
        fireEvent.change(fileTextArea, {target: {value: "C:\\someFolder"}});
        
        const submitButton = container.querySelector(".btn.af-btn");
        fireEvent.click(submitButton);
        
        await waitFor(() => expect(container.querySelector('.image-gallery__link')).not.toBeNull());
        expect(asFragment()).toMatchSnapshot();
        
        const sizeSelect = container.querySelector("#select_type_size");
        const namesSelect = container.querySelector("#select_type_sort");
        fireEvent.change(sizeSelect, {target: {value: "256px"}});
        fireEvent.change(namesSelect, {target: {value: "Alphabetic asc"}});
        
        fireEvent.click(submitButton);
        
        await waitFor(() => expect(container.querySelector('.image-container-width__normal')).not.toBeNull());
        
        await act(async() => new Promise((r) => setTimeout(r, 6000)));
        await waitFor(() => expect(nbFetchCalls).toBeGreaterThanOrEqual(3));

        expect(asFragment()).toMatchSnapshot();
    });
    
    test('Should handle error on bad request', async () => {
        const {container, asFragment} = render(<Router basename="/"><Gallery fetchFunction={fetchKO} /></Router>);

        expect(asFragment()).toMatchSnapshot();

        const fileTextArea = container.querySelector(".af-form__input-text");
        fireEvent.change(fileTextArea, {target: {value: "C:\\someFolder"}});

        const submitButton = container.querySelector(".btn.af-btn");
        fireEvent.click(submitButton);
        
        await waitFor(() => expect(container.querySelector('.gallery__error-message')).not.toBeNull());
        expect(asFragment()).toMatchSnapshot();
    });
});

describe.each([
    {sortName: "Recent to old", files: sortFiles, expected: [sortFiles[3], sortFiles[2], sortFiles[1], sortFiles[0]]},
    {sortName: "Old to recent", files: sortFiles, expected: sortFiles},
    {sortName: "Alphabetic asc", files: sortFiles, expected: sortFiles},
    {sortName: "Alphabetic desc", files: sortFiles, expected: [sortFiles[3], sortFiles[2], sortFiles[1], sortFiles[0]]}
])("Check files sorting order", ({sortName, files, expected}) =>  {
    test(`Sort by order: ${sortName}`, () => {
        const result = sortItems(sortName, files);
        expect(result).toEqual(expected);
    });
});