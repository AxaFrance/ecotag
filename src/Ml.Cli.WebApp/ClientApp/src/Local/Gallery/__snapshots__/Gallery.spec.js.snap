// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check images display Should handle error on bad request 1`] = `
<DocumentFragment>
  <div
    class="af-header"
  >
    <div
      class="container af-header__wrapper"
    >
      <header
        class="af-header__content"
        role="banner"
      >
        <div
          class="af-header__name"
        >
          <a
            class="af-logo"
            href="/"
          >
            <img
              alt="AXA Logo"
              class="af-logo__brand"
              src="logo-axa.svg"
            />
          </a>
          <h2
            class="af-header__title"
          >
            ML-CLI
            <span
              class="af-header__subtitle"
            >
              Made by AXA
            </span>
          </h2>
        </div>
      </header>
    </div>
  </div>
  <div
    class="af-title-bar af-title-bar--backhome af-title-bar--no-space"
  >
    <div
      class="container-fluid container af-title-bar__wrapper"
    >
      <a
        class="btn af-btn--circle"
        href="/"
        title="Home page"
      >
        <i
          class="glyphicon glyphicon-home"
        />
      </a>
      <h1
        class="af-title-bar__title"
      >
        Image Gallery
      </h1>
    </div>
  </div>
  <div
    class="gallery__header"
  >
    <div
      class="gallery__options-container-left"
    >
      <p
        class="tabs__title"
      >
        Directory name:
      </p>
      <input
        class="af-form__input-text"
        id="text_area_stringsMatcherModifier"
        name="TextAreaStringsMatcherModifier"
        type="text"
        value=""
      />
      <p
        class="tabs__title"
      >
        Sort by:
      </p>
      <div
        class="af-form__select-container"
      >
        <select
          class="af-form__input-select"
          id="select_type_sort"
          name="SelectTypeSort"
        >
          <option
            value="Recent to old"
          >
            Recent to old
          </option>
          <option
            value="Old to recent"
          >
            Old to recent
          </option>
          <option
            value="Alphabetic asc"
          >
            Alphabetic asc
          </option>
          <option
            value="Alphabetic desc"
          >
            Alphabetic desc
          </option>
        </select>
        <span
          aria-controls="select_type_sort"
          class="glyphicon glyphicon-menu-down"
        />
      </div>
      <p
        class="tabs__title"
      >
        Size:
      </p>
      <div
        class="af-form__select-container"
      >
        <select
          class="af-form__input-select"
          id="select_type_size"
          name="SelectTypeSize"
        >
          <option
            value="64px"
          >
            64px
          </option>
          <option
            value="128px"
          >
            128px
          </option>
          <option
            value="256px"
          >
            256px
          </option>
          <option
            value="512px"
          >
            512px
          </option>
        </select>
        <span
          aria-controls="select_type_size"
          class="glyphicon glyphicon-menu-down"
        />
      </div>
    </div>
    <div
      class="gallery__options-container-right"
    >
      <button
        class="btn af-btn"
      >
        Apply filters
      </button>
    </div>
  </div>
  <h3
    class="gallery__info-message"
  >
    Please provide a directory path
  </h3>
  <div
    class="image-gallery__container"
  />
  <div
    class="Toastify"
  />
</DocumentFragment>
`;

exports[`Check images display Should handle error on bad request 2`] = `
<DocumentFragment>
  <div
    class="af-header"
  >
    <div
      class="container af-header__wrapper"
    >
      <header
        class="af-header__content"
        role="banner"
      >
        <div
          class="af-header__name"
        >
          <a
            class="af-logo"
            href="/"
          >
            <img
              alt="AXA Logo"
              class="af-logo__brand"
              src="logo-axa.svg"
            />
          </a>
          <h2
            class="af-header__title"
          >
            ML-CLI
            <span
              class="af-header__subtitle"
            >
              Made by AXA
            </span>
          </h2>
        </div>
      </header>
    </div>
  </div>
  <div
    class="af-title-bar af-title-bar--backhome af-title-bar--no-space"
  >
    <div
      class="container-fluid container af-title-bar__wrapper"
    >
      <a
        class="btn af-btn--circle"
        href="/"
        title="Home page"
      >
        <i
          class="glyphicon glyphicon-home"
        />
      </a>
      <h1
        class="af-title-bar__title"
      >
        Image Gallery
      </h1>
    </div>
  </div>
  <div
    class="gallery__header"
  >
    <div
      class="gallery__options-container-left"
    >
      <p
        class="tabs__title"
      >
        Directory name:
      </p>
      <input
        class="af-form__input-text"
        id="text_area_stringsMatcherModifier"
        name="TextAreaStringsMatcherModifier"
        type="text"
        value="C:\\\\someFolder"
      />
      <p
        class="tabs__title"
      >
        Sort by:
      </p>
      <div
        class="af-form__select-container"
      >
        <select
          class="af-form__input-select"
          id="select_type_sort"
          name="SelectTypeSort"
        >
          <option
            value="Recent to old"
          >
            Recent to old
          </option>
          <option
            value="Old to recent"
          >
            Old to recent
          </option>
          <option
            value="Alphabetic asc"
          >
            Alphabetic asc
          </option>
          <option
            value="Alphabetic desc"
          >
            Alphabetic desc
          </option>
        </select>
        <span
          aria-controls="select_type_sort"
          class="glyphicon glyphicon-menu-down"
        />
      </div>
      <p
        class="tabs__title"
      >
        Size:
      </p>
      <div
        class="af-form__select-container"
      >
        <select
          class="af-form__input-select"
          id="select_type_size"
          name="SelectTypeSize"
        >
          <option
            value="64px"
          >
            64px
          </option>
          <option
            value="128px"
          >
            128px
          </option>
          <option
            value="256px"
          >
            256px
          </option>
          <option
            value="512px"
          >
            512px
          </option>
        </select>
        <span
          aria-controls="select_type_size"
          class="glyphicon glyphicon-menu-down"
        />
      </div>
    </div>
    <div
      class="gallery__options-container-right"
    >
      <button
        class="btn af-btn"
      >
        Apply filters
      </button>
    </div>
  </div>
  <h3
    class="gallery__error-message"
  >
    New files fetch on options submission failed
  </h3>
  <div
    class="image-gallery__container"
  />
  <div
    class="Toastify"
  />
</DocumentFragment>
`;

exports[`Check images display Should render Gallery page correctly, resize images and sort by names 1`] = `
<DocumentFragment>
  <div
    class="af-header"
  >
    <div
      class="container af-header__wrapper"
    >
      <header
        class="af-header__content"
        role="banner"
      >
        <div
          class="af-header__name"
        >
          <a
            class="af-logo"
            href="/"
          >
            <img
              alt="AXA Logo"
              class="af-logo__brand"
              src="logo-axa.svg"
            />
          </a>
          <h2
            class="af-header__title"
          >
            ML-CLI
            <span
              class="af-header__subtitle"
            >
              Made by AXA
            </span>
          </h2>
        </div>
      </header>
    </div>
  </div>
  <div
    class="af-title-bar af-title-bar--backhome af-title-bar--no-space"
  >
    <div
      class="container-fluid container af-title-bar__wrapper"
    >
      <a
        class="btn af-btn--circle"
        href="/"
        title="Home page"
      >
        <i
          class="glyphicon glyphicon-home"
        />
      </a>
      <h1
        class="af-title-bar__title"
      >
        Image Gallery
      </h1>
    </div>
  </div>
  <div
    class="gallery__header"
  >
    <div
      class="gallery__options-container-left"
    >
      <p
        class="tabs__title"
      >
        Directory name:
      </p>
      <input
        class="af-form__input-text"
        id="text_area_stringsMatcherModifier"
        name="TextAreaStringsMatcherModifier"
        type="text"
        value=""
      />
      <p
        class="tabs__title"
      >
        Sort by:
      </p>
      <div
        class="af-form__select-container"
      >
        <select
          class="af-form__input-select"
          id="select_type_sort"
          name="SelectTypeSort"
        >
          <option
            value="Recent to old"
          >
            Recent to old
          </option>
          <option
            value="Old to recent"
          >
            Old to recent
          </option>
          <option
            value="Alphabetic asc"
          >
            Alphabetic asc
          </option>
          <option
            value="Alphabetic desc"
          >
            Alphabetic desc
          </option>
        </select>
        <span
          aria-controls="select_type_sort"
          class="glyphicon glyphicon-menu-down"
        />
      </div>
      <p
        class="tabs__title"
      >
        Size:
      </p>
      <div
        class="af-form__select-container"
      >
        <select
          class="af-form__input-select"
          id="select_type_size"
          name="SelectTypeSize"
        >
          <option
            value="64px"
          >
            64px
          </option>
          <option
            value="128px"
          >
            128px
          </option>
          <option
            value="256px"
          >
            256px
          </option>
          <option
            value="512px"
          >
            512px
          </option>
        </select>
        <span
          aria-controls="select_type_size"
          class="glyphicon glyphicon-menu-down"
        />
      </div>
    </div>
    <div
      class="gallery__options-container-right"
    >
      <button
        class="btn af-btn"
      >
        Apply filters
      </button>
    </div>
  </div>
  <h3
    class="gallery__info-message"
  >
    Please provide a directory path
  </h3>
  <div
    class="image-gallery__container"
  />
  <div
    class="Toastify"
  />
</DocumentFragment>
`;

exports[`Check images display Should render Gallery page correctly, resize images and sort by names 2`] = `
<DocumentFragment>
  <div
    class="af-header"
  >
    <div
      class="container af-header__wrapper"
    >
      <header
        class="af-header__content"
        role="banner"
      >
        <div
          class="af-header__name"
        >
          <a
            class="af-logo"
            href="/"
          >
            <img
              alt="AXA Logo"
              class="af-logo__brand"
              src="logo-axa.svg"
            />
          </a>
          <h2
            class="af-header__title"
          >
            ML-CLI
            <span
              class="af-header__subtitle"
            >
              Made by AXA
            </span>
          </h2>
        </div>
      </header>
    </div>
  </div>
  <div
    class="af-title-bar af-title-bar--backhome af-title-bar--no-space"
  >
    <div
      class="container-fluid container af-title-bar__wrapper"
    >
      <a
        class="btn af-btn--circle"
        href="/"
        title="Home page"
      >
        <i
          class="glyphicon glyphicon-home"
        />
      </a>
      <h1
        class="af-title-bar__title"
      >
        Image Gallery
      </h1>
    </div>
  </div>
  <div
    class="gallery__header"
  >
    <div
      class="gallery__options-container-left"
    >
      <p
        class="tabs__title"
      >
        Directory name:
      </p>
      <input
        class="af-form__input-text"
        id="text_area_stringsMatcherModifier"
        name="TextAreaStringsMatcherModifier"
        type="text"
        value="C:\\\\someFolder"
      />
      <p
        class="tabs__title"
      >
        Sort by:
      </p>
      <div
        class="af-form__select-container"
      >
        <select
          class="af-form__input-select"
          id="select_type_sort"
          name="SelectTypeSort"
        >
          <option
            value="Recent to old"
          >
            Recent to old
          </option>
          <option
            value="Old to recent"
          >
            Old to recent
          </option>
          <option
            value="Alphabetic asc"
          >
            Alphabetic asc
          </option>
          <option
            value="Alphabetic desc"
          >
            Alphabetic desc
          </option>
        </select>
        <span
          aria-controls="select_type_sort"
          class="glyphicon glyphicon-menu-down"
        />
      </div>
      <p
        class="tabs__title"
      >
        Size:
      </p>
      <div
        class="af-form__select-container"
      >
        <select
          class="af-form__input-select"
          id="select_type_size"
          name="SelectTypeSize"
        >
          <option
            value="64px"
          >
            64px
          </option>
          <option
            value="128px"
          >
            128px
          </option>
          <option
            value="256px"
          >
            256px
          </option>
          <option
            value="512px"
          >
            512px
          </option>
        </select>
        <span
          aria-controls="select_type_size"
          class="glyphicon glyphicon-menu-down"
        />
      </div>
    </div>
    <div
      class="gallery__options-container-right"
    >
      <button
        class="btn af-btn"
      >
        Apply filters
      </button>
    </div>
  </div>
  <div
    class="image-gallery__container"
  >
    <div
      class="image-gallery__link"
    >
      <a
        href="api/files/Qzpcc29tZUZvbGRlclxzb21lUGRmLnBkZg=="
        rel="noopener noreferrer"
        target="_blank"
      >
        <div
          class="image-gallery__image-container image-container-width__medium"
        >
          <img
            alt="C:\\\\someFolder\\\\somePdf.pdf"
            class="image-gallery__image"
            src="file-pdf.svg"
          />
        </div>
        <div
          class="image-gallery__filename-container filename-width__medium"
        >
          <div
            class="image-gallery__filename"
          >
            somePdf.pdf
          </div>
        </div>
      </a>
    </div>
    <div
      class="image-gallery__link"
    >
      <a
        href="api/files/Qzpcc29tZUZvbGRlclxzb21lSW1hZ2UucG5n"
        rel="noopener noreferrer"
        target="_blank"
      >
        <div
          class="image-gallery__image-container image-container-width__medium"
        >
          <img
            alt="C:\\\\someFolder\\\\someImage.png"
            class="image-gallery__image"
            src="api/files/Qzpcc29tZUZvbGRlclxzb21lSW1hZ2UucG5n"
          />
        </div>
        <div
          class="image-gallery__filename-container filename-width__medium"
        >
          <div
            class="image-gallery__filename"
          >
            someImage.png
          </div>
        </div>
      </a>
    </div>
    <div
      class="image-gallery__link"
    >
      <a
        download="compare-licenses-file-2.json"
        href="api/files/Qzpcc29tZUZvbGRlclxjb21wYXJlLWxpY2Vuc2VzLWZpbGUtMi5qc29u"
      >
        <div
          class="image-gallery__image-container image-container-width__medium"
        >
          <img
            alt="C:\\\\someFolder\\\\compare-licenses-file-2.json_default-icon"
            class="image-gallery__image"
            src="file.svg"
          />
        </div>
        <div
          class="image-gallery__filename-container filename-width__medium"
        >
          <div
            class="image-gallery__filename"
          >
            compare-licenses-file-2.json
          </div>
        </div>
      </a>
    </div>
    <div
      class="image-gallery__link"
    >
      <a
        download="compare-licenses-file-1.json"
        href="api/files/Qzpcc29tZUZvbGRlclxjb21wYXJlLWxpY2Vuc2VzLWZpbGUtMS5qc29u"
      >
        <div
          class="image-gallery__image-container image-container-width__medium"
        >
          <img
            alt="C:\\\\someFolder\\\\compare-licenses-file-1.json_default-icon"
            class="image-gallery__image"
            src="file.svg"
          />
        </div>
        <div
          class="image-gallery__filename-container filename-width__medium"
        >
          <div
            class="image-gallery__filename"
          >
            compare-licenses-file-1.json
          </div>
        </div>
      </a>
    </div>
  </div>
  <div
    class="Toastify"
  />
</DocumentFragment>
`;

exports[`Check images display Should render Gallery page correctly, resize images and sort by names 3`] = `
<DocumentFragment>
  <div
    class="af-header"
  >
    <div
      class="container af-header__wrapper"
    >
      <header
        class="af-header__content"
        role="banner"
      >
        <div
          class="af-header__name"
        >
          <a
            class="af-logo"
            href="/"
          >
            <img
              alt="AXA Logo"
              class="af-logo__brand"
              src="logo-axa.svg"
            />
          </a>
          <h2
            class="af-header__title"
          >
            ML-CLI
            <span
              class="af-header__subtitle"
            >
              Made by AXA
            </span>
          </h2>
        </div>
      </header>
    </div>
  </div>
  <div
    class="af-title-bar af-title-bar--backhome af-title-bar--no-space"
  >
    <div
      class="container-fluid container af-title-bar__wrapper"
    >
      <a
        class="btn af-btn--circle"
        href="/"
        title="Home page"
      >
        <i
          class="glyphicon glyphicon-home"
        />
      </a>
      <h1
        class="af-title-bar__title"
      >
        Image Gallery
      </h1>
    </div>
  </div>
  <div
    class="gallery__header"
  >
    <div
      class="gallery__options-container-left"
    >
      <p
        class="tabs__title"
      >
        Directory name:
      </p>
      <input
        class="af-form__input-text"
        id="text_area_stringsMatcherModifier"
        name="TextAreaStringsMatcherModifier"
        type="text"
        value="C:\\\\someFolder"
      />
      <p
        class="tabs__title"
      >
        Sort by:
      </p>
      <div
        class="af-form__select-container"
      >
        <select
          class="af-form__input-select"
          id="select_type_sort"
          name="SelectTypeSort"
        >
          <option
            value="Recent to old"
          >
            Recent to old
          </option>
          <option
            value="Old to recent"
          >
            Old to recent
          </option>
          <option
            value="Alphabetic asc"
          >
            Alphabetic asc
          </option>
          <option
            value="Alphabetic desc"
          >
            Alphabetic desc
          </option>
        </select>
        <span
          aria-controls="select_type_sort"
          class="glyphicon glyphicon-menu-down"
        />
      </div>
      <p
        class="tabs__title"
      >
        Size:
      </p>
      <div
        class="af-form__select-container"
      >
        <select
          class="af-form__input-select"
          id="select_type_size"
          name="SelectTypeSize"
        >
          <option
            value="64px"
          >
            64px
          </option>
          <option
            value="128px"
          >
            128px
          </option>
          <option
            value="256px"
          >
            256px
          </option>
          <option
            value="512px"
          >
            512px
          </option>
        </select>
        <span
          aria-controls="select_type_size"
          class="glyphicon glyphicon-menu-down"
        />
      </div>
    </div>
    <div
      class="gallery__options-container-right"
    >
      <button
        class="btn af-btn"
      >
        Apply filters
      </button>
    </div>
  </div>
  <div
    class="image-gallery__container"
  >
    <div
      class="image-gallery__link"
    >
      <a
        download="compare-licenses-file-1.json"
        href="api/files/Qzpcc29tZUZvbGRlclxjb21wYXJlLWxpY2Vuc2VzLWZpbGUtMS5qc29u"
      >
        <div
          class="image-gallery__image-container image-container-width__normal"
        >
          <img
            alt="C:\\\\someFolder\\\\compare-licenses-file-1.json_default-icon"
            class="image-gallery__image"
            src="file.svg"
          />
        </div>
        <div
          class="image-gallery__filename-container filename-width__normal"
        >
          <div
            class="image-gallery__filename"
          >
            compare-licenses-file-1.json
          </div>
        </div>
      </a>
    </div>
    <div
      class="image-gallery__link"
    >
      <a
        download="compare-licenses-file-2.json"
        href="api/files/Qzpcc29tZUZvbGRlclxjb21wYXJlLWxpY2Vuc2VzLWZpbGUtMi5qc29u"
      >
        <div
          class="image-gallery__image-container image-container-width__normal"
        >
          <img
            alt="C:\\\\someFolder\\\\compare-licenses-file-2.json_default-icon"
            class="image-gallery__image"
            src="file.svg"
          />
        </div>
        <div
          class="image-gallery__filename-container filename-width__normal"
        >
          <div
            class="image-gallery__filename"
          >
            compare-licenses-file-2.json
          </div>
        </div>
      </a>
    </div>
    <div
      class="image-gallery__link"
    >
      <a
        href="api/files/Qzpcc29tZUZvbGRlclxzb21lSW1hZ2UucG5n"
        rel="noopener noreferrer"
        target="_blank"
      >
        <div
          class="image-gallery__image-container image-container-width__normal"
        >
          <img
            alt="C:\\\\someFolder\\\\someImage.png"
            class="image-gallery__image"
            src="api/files/Qzpcc29tZUZvbGRlclxzb21lSW1hZ2UucG5n"
          />
        </div>
        <div
          class="image-gallery__filename-container filename-width__normal"
        >
          <div
            class="image-gallery__filename"
          >
            someImage.png
          </div>
        </div>
      </a>
    </div>
    <div
      class="image-gallery__link"
    >
      <a
        href="api/files/Qzpcc29tZUZvbGRlclxzb21lUGRmLnBkZg=="
        rel="noopener noreferrer"
        target="_blank"
      >
        <div
          class="image-gallery__image-container image-container-width__normal"
        >
          <img
            alt="C:\\\\someFolder\\\\somePdf.pdf"
            class="image-gallery__image"
            src="file-pdf.svg"
          />
        </div>
        <div
          class="image-gallery__filename-container filename-width__normal"
        >
          <div
            class="image-gallery__filename"
          >
            somePdf.pdf
          </div>
        </div>
      </a>
    </div>
  </div>
  <div
    class="Toastify"
  />
</DocumentFragment>
`;
